docs/markdown/Installing.md

```diff
@@ -133,23 +133,55 @@ command in the build tree:
 $ meson install --no-rebuild --only-changed
 ```
 
-## Finer control over install locations
-
-Sometimes it is necessary to only install a subsection of output files
-or install them in different directories. This can be done by
-specifying `install_dir` as an array rather than a single string. The
-array must have as many items as there are outputs and each entry
-specifies how the corresponding output file should be installed. For
-example:
-
-```meson
-custom_target(...
-    output: ['file1', 'file2', 'file3'],
-    install_dir: ['path1', false, 'path3'],
-    ...
-)
-```
-
-In this case `file1` would be installed to `/prefix/path1/file1`,
-`file2` would not be installed at all and `file3` would be installed
-to `/prefix/path3/file3'.
+## Installation tags
+
+*Since 0.60.0*
+
+It is possible to install only a subset of the installable files using
+`meson install --tags tag1,tag2` command line. When `--tags` is specified, only
+files that have been tagged with one of the tags are going to be installed.
+
+This is intended to be used by packagers (e.g. distributions) who typically
+want to split `libfoo`, `libfoo-dev` and `libfoo-doc` packages. Instead of
+duplicating the list of installed files per category in each packaging system,
+it can be maintained in a single place, directly in upstream `meson.build` files.
```

Comment:
This seems to assume that all distros will agree about what package a file ends up in.  I'm not sure that's true.
 
I'm not sure I understand how this is supposed to be used by a distro packaging tool? Run the install step with each of a set of tags into a clean DESTDIR, and package the files which end up there?  How is the tool supposed to learn the set of tags needed to span all installed files?

Reply:
Note that here I mean packagers in a larger sense than just distro. For example in my current project we deploy the runtime as a tarball we extract on devices, that "packaging" is easier to do if I can just tell meson to install only runtime files in my DESTDIR, then I can just tar that dir and copy on devices.

I'm not expecting distro like debian to adopt this system any time soon, they already have all their tooling based on maintaining list of files for each binary package. But if they do, yes they would run install into different DESTDIRs for runtime, devel, etc.

I defined a set of well known tags in the doc, I think most projects should stick that that list. But I also saw use-cases for more specific projects (not typically packaged in distro) that would want to define their own split.